editors/vim/Portfile

```diff
@@ -107,50 +107,31 @@ variant perl description {Enable Perl scripting} {
     depends_lib-append      path:bin/perl:perl5
 }
 
-variant python27 description {Enable Python scripting} {
-    configure.args-append   --enable-pythoninterp --with-python-command=${prefix}/bin/python2.7
-    patchfiles-append       patch-python.diff
-    depends_lib-append      port:python27
-
-    use_autoconf yes
-    # Overwriting autoconf.cmd above removes dependency, add it again
-    depends_build-append port:autoconf
-}
-variant python36 conflicts python37 python38 python39 description {Enable Python scripting} {
-    configure.args-append   --enable-python3interp --with-python3-command=${prefix}/bin/python3.6
-    patchfiles-append       patch-python3.diff
-    depends_lib-append      port:python36
-
-    use_autoconf yes
-    # Overwriting autoconf.cmd above removes dependency, add it again
-    depends_build-append port:autoconf
-}
-variant python37 conflicts python36 python38 python39 description {Enable Python scripting} {
-    configure.args-append   --enable-python3interp --with-python3-command=${prefix}/bin/python3.7
-    patchfiles-append       patch-python3.diff
-    depends_lib-append      port:python37
-
-    use_autoconf yes
-    # Overwriting autoconf.cmd above removes dependency, add it again
-    depends_build-append port:autoconf
-}
-variant python38 conflicts python36 python37 python39 description {Enable Python scripting} {
-    configure.args-append   --enable-python3interp --with-python3-command=${prefix}/bin/python3.8
-    patchfiles-append       patch-python3.diff
-    depends_lib-append      port:python38
-
-    use_autoconf yes
-    # Overwriting autoconf.cmd above removes dependency, add it again
-    depends_build-append port:autoconf
-}
-variant python39 conflicts python36 python37 python38 description {Enable Python scripting} {
-    configure.args-append   --enable-python3interp --with-python3-command=${prefix}/bin/python3.9
-    patchfiles-append       patch-python3.diff
-    depends_lib-append      port:python39
-
-    use_autoconf yes
-    # Overwriting autoconf.cmd above removes dependency, add it again
-    depends_build-append port:autoconf
+set pythons_suffixes {27 36 37 38 39}
+
+set pythons_ports {}
+foreach s ${pythons_suffixes} {
+    lappend pythons_ports python${s}
+}
+
+foreach s ${pythons_suffixes} {
+    set p python${s}
+    set v [string index ${s} 0].[string range ${s} 1 end]
+    set i [lsearch -exact ${pythons_ports} ${p}]
+    set c [lreplace ${pythons_ports} ${i} ${i}]
+    variant ${p} description "Enable Python scripting" conflicts {*}${c} "
+        if {${s} < 300} {
```

Comment:
Isn’t this going to be true for Python 3 before 3.10, for example, `39` for 3.9, causing those variants to break?

Reply:
Thanks for flagging this, I had noticed it and thought I pushed the fix. Apparently not.

You can find the correction [here](https://github.com/macports/macports-ports/pull/14084). Apologies for breaking things.